src/librustdoc/clean/mod.rs

```diff
@@ -2792,10 +2791,11 @@ fn clean_maybe_renamed_item<'tcx>(
     use hir::ItemKind;
 
     let def_id = item.owner_id.to_def_id();
-    let mut name = renamed.unwrap_or_else(|| {
-        // FIXME: using kw::Empty is a bit of a hack
-        cx.tcx.hir_opt_name(item.hir_id()).unwrap_or(kw::Empty)
-    });
+    let mut name = if renamed.is_some() {
+        renamed
+    } else {
+        cx.tcx.hir_opt_name(item.hir_id())
+    };
```

Comment:
No need to keep an `Option` around, right? If so:

```suggestion
    let name = renamed.unwrap_or_else(|| cx.tcx.hir_opt_name(item.hir_id()).unwrap());
```

Reply:
That doesn't work because he `hir_opt_name` can return `None`. So it must be an `Option`, and then lower down we only unwrap it for certain item kinds that always have an ident.